File: src/main/java/controls/lambdas/Example1.java
package controls.lambdas;

import java.util.function.Supplier;

public class Example1 {
    public void useIt() {
        System.out.println(condition(true, () -> "Hello Java 8"));
    }

    public String condition(boolean condition, Supplier<String> f) {
        if (condition) {
            return f.get();
        } else {
            return "false";
        }
    }
}

File: src/main/scala/examples/lambdas/Example1.scala
package examples.lambdas

class Example1 {
  def useIt(): Unit = {
    println(condition(true, () => "Hello Java 8"))
  }

  def condition(condition: Boolean, f: () => String): String = {
    if (condition)
      f.apply()
    else
      "false"
  }
}
